test/tests/functional/pbs_test_tpp.py

```diff
@@ -383,38 +383,31 @@ def common_setup(self):
         comm objects : self.comm2, self.comm3
         comm shortnames : self.hostD, self.hostE
         """
-        mom_list = copy.copy(self.moms.values())
-        comm_list = copy.copy(self.comms.values())
-        _server = self.server.shortname
-        skip_test = [False if _server == mom.shortname else
-                     True for mom in self.moms.values()] and \
-                    [False if _server == comm.shortname else True for
-                     comm in self.comms.values()]
-        if skip_test[0]:
-            self.skipTest("Mom and Comm should be on server host")
-        for i, mom in enumerate(self.moms.values()):
-            if mom.shortname == self.server.shortname:
-                self.momA = mom
-                self.hostA = self.momA.shortname
-                del mom_list[i]
-                break
-        for j, comm in enumerate(self.comms.values()):
-            if comm.shortname == self.server.shortname:
-                del comm_list[j]
+        mom_list = []
+        comm_list = []
+        for mom in self.moms.values():
+            mom_list.append(mom.shortname)
+        for comm in self.comms.values():
+            comm_list.append(comm.shortname)
```

Comment:
If you want, you can use list comprehension here.  Turn this for loop into 
[x.shortname for x in self.comms.values()]

Same for the moms above